src/device-registry/routes/api-v2.js

```diff
@@ -3052,18 +3052,16 @@ router.post(
 );
 router.get(
   "/events",
-  oneOf([
-    query("tenant")
-      .exists()
-      .withMessage("tenant should be provided")
-      .bail()
-      .trim()
-      .toLowerCase()
-      .isIn(["kcca", "airqo", "view"])
-      .withMessage("the tenant value is not among the expected ones"),
-  ]),
   oneOf([
     [
+      query("tenant")
+        .optional()
+        .notEmpty()
+        .withMessage("the tenant should not be empty if provided")
+        .trim()
+        .toLowerCase()
+        .isIn(["kcca", "airqo", "view"])
+        .withMessage("the tenant value is not among the expected ones"),
       query("startTime")
```

Comment:
these input validations for datetime need to be thoroughly reviewed to ensure that any bad input from users are captured way before we make any calls to BQ. cc: @NoahNsimbe @123MwanjeMike